kie-spring-boot/kie-spring-boot-samples/keycloak-kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KeycloakKieServerTest.java

```diff
@@ -210,4 +227,79 @@ public void testProcessStartAndWorkOnUserTask() {
         assertNotNull(processInstance);
         assertEquals(3, processInstance.getState().intValue());        
     }
+    
+    @Test
+    public void testAuthorizedUserOnRestrictedVar() {
+
+        setupClient(BARTLET, BARTLET_PW);
+
+        ProcessServicesClient processClient = kieServicesClient.getServicesClient(ProcessServicesClient.class);
+        
+        // allowed user can start process instance
+        Long processInstanceId = processClient.startProcess(containerId, restrictedVarProcessId, singletonMap("press", Boolean.TRUE));
+        assertNotNull(processInstanceId);
+       
+        abortProcess(processClient, processInstanceId);
+    }
+    
+    @Test
+    public void testNoRestrictedVarViolation() {
+
+        setupClient(JOHN, JOHN_PW);
+
+        ProcessServicesClient processClient = kieServicesClient.getServicesClient(ProcessServicesClient.class);
+        
+        // any user can start that process instance without changing restricted variable value
+        Long processInstanceId = processClient.startProcess(containerId, restrictedVarProcessId);
+        assertNotNull(processInstanceId);
+       
+        abortProcess(processClient, processInstanceId);
+    }
+
+    
+    @Test
+    public void testRestrictedVarViolationByUnauthorizedUser() {
+
+        setupClient(JOHN, JOHN_PW);
+
+        ProcessServicesClient processClient = kieServicesClient.getServicesClient(ProcessServicesClient.class);
+        
+        // unallowed user cannot start process instance
+        assertThatExceptionOfType(KieServicesHttpException.class)
+              .isThrownBy(() -> processClient.startProcess(containerId, restrictedVarProcessId, singletonMap("press", Boolean.TRUE)));
+    }
+    
+    private static boolean isDockerAvailable() {
+        try {
+            DockerClientFactory.instance().client();
+            return true;
+        } catch (Throwable ex) {
+            return false;
+        }
+    }
```

Comment:
Not sure if this is what we really want here. I mean, for instance, I got docker installed on my laptop so this test is executed properly, but I guess it shouldn't be since RHEL8 is not "officially" supported by testcontainers framework. 

Also not sure about the outcome when this gets executed on a RHEL7 as Docker doesn't come out of the box.

So I guess we should make sure this is running fine for both scenarios/OS - What do you think?

Reply:
this way is more generic: if docker is present, tests will work